core/Math/hsGeometry3.h

```diff
@@ -107,6 +107,16 @@ struct PLASMA_DLL hsVector3 {
                          (Z * other.X) - (X * other.Z),
                          (X * other.Y) - (Y * other.X));
     }
+
+    /** Converts this vector a unit vector such that its magnitude is 1 */
+    void normalize() {
+        float length = magnitude();
+        if (length == 0.f || length == 1.f)
```

Comment:
Floating point math is not precise...  I feel like we should have a "close enough to 1.f" rather than "exactly 1.f" check, or re-normalizing may become lossy